src/sentry/taskworker/worker.py

```diff
@@ -48,7 +52,22 @@ class ProcessingResult:
     status: TaskActivationStatus.ValueType
 
 
-AT_MOST_ONCE_TIMEOUT = 60 * 60 * 24  # 1 day
+@contextlib.contextmanager
+def timeout_alarm(seconds: int, handler: Callable[[int, Any], None]):
```

Comment:
since the actual handler doesn't care about either of its arguments it can be `Callable[[object, object], None]` and this should `-> Generator[None]`